events/notifications.go

```diff
@@ -1,71 +1,340 @@
+	"github.com/flanksource/incident-commander/teams"
+	"github.com/flanksource/incident-commander/utils"
 	"github.com/google/uuid"
-	"gorm.io/gorm"
 )
 
-func addNotificationEvent(ctx *api.Context, tx *gorm.DB, responder api.Responder) error {
-	var notifications []api.Notification
-	if err := tx.WithContext(ctx).Where("team_id = ? AND deleted_at IS NULL", responder.TeamID).Find(&notifications).Error; err != nil {
+type NotificationEventProperties struct {
+	ID               string `json:"id"`                          // Resource id. depends what it is based on the original event.
+	EventName        string `json:"event_name"`                  // The name of the original event this notification is for.
+	PersonID         string `json:"person_id,omitempty"`         // The person recipient.
+	TeamID           string `json:"team_id,omitempty"`           // The team recipient.
+	NotificationName string `json:"notification_name,omitempty"` // Name of the notification of a team or a custom service of the notification.
+	NotificationID   string `json:"notification_id,omitempty"`   // ID of the notification.
+}
+
+// NotificationTemplate holds in data for notification
+// that'll be used by struct templater.
+type NotificationTemplate struct {
+	Message    string            `template:"true"`
+	Properties map[string]string `template:"true"`
+}
+
+func (t *NotificationEventProperties) AsMap() map[string]string {
+	m := make(map[string]string)
+	b, _ := json.Marshal(&t)
+	_ = json.Unmarshal(b, &m)
+	return m
+}
+
+func (t *NotificationEventProperties) FromMap(m map[string]string) {
+	b, _ := json.Marshal(m)
+	_ = json.Unmarshal(b, &t)
+}
+
+func publishNotification(ctx *api.Context, event api.Event) error {
+	var props NotificationEventProperties
+	props.FromMap(event.Properties)
+
+	celEnv, err := getEnvForEvent(ctx, props.EventName, event.Properties)
+	if err != nil {
 		return err
 	}
 
-	for _, n := range notifications {
-		if valid, err := notification.IsValid(n.Config.Filter, responder); err != nil {
-			return err
-		} else if !valid {
-			continue
+	templater := template.StructTemplater{
+		Values:         celEnv,
+		ValueFunctions: true,
+		DelimSets: []template.Delims{
+			{Left: "{{", Right: "}}"},
+			{Left: "$(", Right: ")"},
+		},
+	}
+
+	notification, err := pkgNotification.GetNotification(ctx, props.NotificationID)
+	if err != nil {
+		return err
+	}
+
+	data := NotificationTemplate{
+		Message:    notification.Template,
+		Properties: notification.Properties,
+	}
+
+	if err := templater.Walk(&data); err != nil {
+		return fmt.Errorf("error templating notification: %w", err)
+	}
+
+	if props.PersonID != "" {
+		var emailAddress string
+		if err := ctx.DB().Model(&models.Person{}).Select("email").Where("id = ?", props.PersonID).Find(&emailAddress).Error; err != nil {
+			return fmt.Errorf("failed to get email of person(id=%s); %v", props.PersonID, err)
+		}
+
+		url := fmt.Sprintf("smtp://%s:%s@%s:%s/?auth=Plain&FromAddress=%s&ToAddresses=%s",
```

Comment:
We can't add the smtp details into the event message - I would just have a `email` event

Reply:
@moshloop We are not adding it to the event. We're publishing it directly.